lib/ddtrace/contrib/ethon/multi_patch.rb

```diff
@@ -19,19 +19,48 @@ def add(easy)
             handles = super(easy)
             return handles if handles.nil? || !tracer_enabled?
 
-            easy.datadog_before_request
+            easy.datadog_before_request(parent_span: datadog_multi_span)
```

Comment:
Is it possible there's a delay between `#add` and `#perform`? When the span is created, it's auto created with the start time. I'm thinking about the possibility where someone creates a `Multi` request with `Easy` children, but then delays a while before actually executing it; could that cause the timing to be recorded incorrectly?

Reply:
It is possible, more precise solution would probably require storing some state on `Multi` object. The complexity comes from the fact that extra `Easy` object can be added to `Multi` during the execution of `perform` (in easy complete callback, this is how Typhoeus implementation works). Current solution will work for Typhoeus since it always adding `easy` objects to `multi` right before executing it.
The same issue as you described can happen with `Easy` span because it is started when easy object is added to `multi` (which is not necessarily executing yet).
We can probably solve the issue by storing some flag on `multi` saying whether it is performing now or not.
Please let me know what you think. I'll be away from computer for one more week and will be able to work on this feature afterwards.

Reply:
Might be worth it, as long as there isn't too much coupling between `Easy` and `Multi` instrumentation.

Is there any way to create the span at `perform` time for `Easy`? Even if that means storing other state on `Easy` initialization beforehand? Given these objects sound idempotent anyway, some state would be permissible.

Reply:
It turned out extra state is not needed. `Easy` handles are stored on `Multi` instances so I can go through them and start child spans. When adding `easy` to `multi`, I can check if there is a span on `multi` which is an indication of `multi`'s `perform` execution in progress.